project/utils/automata_utils.py

```diff
@@ -25,13 +30,77 @@ def transform_regex_to_dfa(regex_str: str) -> DeterministicFiniteAutomaton:
     -------
     dfa: DeterministicFiniteAutomaton
         Minimal DFA built on given regular expression
+
+    Raises
+    ------
+    AutomataException
+        If the incorrect regular expression is passed to the function
     """
 
     try:
         regex = Regex(regex_str)
-    except MisformedRegexError as e:
+    except MisformedRegexError:
         raise AutomataException(f"Invalid regular expression")
```

Comment:
Если вы трансформируете `Exception`, то стоит делать это через `Exception Chaining` (см. [тут](https://docs.python.org/3/tutorial/errors.html#exception-chaining))
```suggestion
    except MisformedRegexError as exc:
        raise AutomataException(f"Invalid regular expression") from exc
```